### PR TITLE
Add basic string escaping

### LINKED ISSUE #29: Braces not escaped in type strings
`map[something]struct{}` (and similar) fail to render because the `{` and `}` are not escaped in the output.
`strconv.Quote` should be used on all type strings to prevent this.

### PR DESCRIPTION
**Pull request by bradleyjkemp**
Fixes #29

### CODE REVIEW

File: escape_test.go [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+	{"Hello world", "Hello world"},
+
+	// double quotes are escaped
+	{"\"Hello world\"", "\\\"Hello world\\\""},
+
+	// brackets not escaped
+	{"map[string]bool", "map[string]bool"},
+
+	// braces escaped
+	{"map[string]struct{}", "map[string]struct\\{\\}"},
```
Comment (piotrkowalczuk) [inline escape_test.go]: Similar issue that I have encountered: `interface {}(nil)`.

**Comment by piotrkowalczuk** [comment]
@bradleyjkemp I would love to see this change merged into master.

**Comment by bradleyjkemp** [comment]
Hey @piotrkowalczuk, thanks for the prod about this. Could you let me know if this fixes the problems you were encountering?

**Comment by piotrkowalczuk** [comment]
@bradleyjkemp Thanks for quick response!

My output looks like 
```Error: bad label format <name> task |{<f0> name | "generate-beggining"} |<f1> anchor|{<f2> err | error(nil)} |{<f3> result | interface {}(nil)}```

Currently I have small hack for it. After passing it through `strings.Replace(buf.String(), "interface {}(nil)", "interface(nil)", -1)`. 
`cat /tmp/memviz.dot | dot -Tpng -o /tmp/memviz.png` produces valid image.

**Comment by bradleyjkemp** [comment]
Ah yes, good point: found a missing escapeString() call. PR #35 should fix your issue.

If not then a small example of a datastructure that causes this failure would be very useful.

**Pull request merged**